android/compose/src/main/kotlin/com/boswelja/ephemeris/compose/InfiniteHorizontalPager.kt

```diff
@@ -21,39 +23,45 @@ import androidx.compose.ui.unit.LayoutDirection
 import dev.chrisbanes.snapper.ExperimentalSnapperApi
 import dev.chrisbanes.snapper.rememberLazyListSnapperLayoutInfo
 import dev.chrisbanes.snapper.rememberSnapperFlingBehavior
+import kotlinx.coroutines.flow.filterNot
 
 @OptIn(ExperimentalSnapperApi::class)
 @Composable
-public fun InfiniteHorizontalPager(
+internal fun InfiniteHorizontalPager(
     modifier: Modifier = Modifier,
     state: InfinitePagerState = rememberInfinitePagerState(),
     contentPadding: PaddingValues = PaddingValues(),
     maxItemFling: Int = 1,
     content: @Composable (page: Int) -> Unit
 ) {
-    val internalLazyState = rememberLazyListState(initialFirstVisibleItemIndex = state.internalStartPage)
     val layoutInfo = rememberLazyListSnapperLayoutInfo(
-        lazyListState = internalLazyState,
+        lazyListState = state.lazyListState,
         endContentPadding = contentPadding.calculateEndPadding(LayoutDirection.Ltr)
     )
 
     // Update the pager state page
-    LaunchedEffect(internalLazyState.isScrollInProgress) {
-        if (!internalLazyState.isScrollInProgress) {
-            state.page = layoutInfo.currentItem?.index ?: state.internalStartPage
-        }
+    LaunchedEffect(state.lazyListState) {
+        snapshotFlow { state.lazyListState.isScrollInProgress }
+            .filterNot { it }
+            .collect {
+                state.page = state.calculatePageFromInternal(
+                    layoutInfo.currentItem?.index ?: state.internalPageOffset
+                )
+            }
     }
 
     var pagerHeight by remember { mutableStateOf(0) }
 
     // Switch the Modifier used to animate height once pagerHeight has had it's initial update
     val layoutModifier = if (pagerHeight > 0) {
-        Modifier.animateContentSize().then(modifier)
+        Modifier
+            .animateContentSize()
```

Comment:
this is so unfair...

Reply:
I'm not super happy with having to switch Modifiers based on the pager height, but `animateContentSize` is terrific
Just don't look down 😆